src/util/records.py

```diff
@@ -1475,7 +1475,7 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
                         """
                         UPDATE {table}
                            SET {column} = (
-                                   SELECT jsonb_object_agg(key, COALESCE((%s::jsonb->>value)::int, value::int))
+                                   SELECT jsonb_object(ARRAY_AGG(key), %s)
```

Comment:
```suggestion
                                   SELECT jsonb_object_agg(key, (jsonb_object(%s::text[], %s::text[])->>value)::int, value::int)
```

Reply:
Yes, this version avoids having literal `{` in the query.

Let's try it over my patch.